### PR TITLE
fix(ip_filter_object): use set type

### LINKED ISSUE #1638: ip_filter_objects reordering without change
## What happened?

We have multiple ip_filter_objects setup like this:
```
    ip_filter_object {
      network = "x.x.x.x/23"
      description = "a ipv4"
    }
    ip_filter_object {
      network     = "x:x:x::/48"
      description = "a ipv6"
    }
    ip_filter_object {
      network     = "x.x.x.x/16"
      description = "b"
    }
    ip_filter_object {
      network     = "x.x.x.x/16"
      description = "c"
    }
    ip_filter_object {
      network = "${var.d}/32"
      description = "d"
    }
```
Most of them are static defined as a hardcoded string. However, we also have one static ip (var.d) that is defined through another resource. Since it's a public static ip address, this one is not changing.

Without changing anything in the terraform configuration, the order if ip_filter_objects gets randomly reshuffled. The content stays the same, only the order changes.
```
          ~ ip_filter_object {
              ~ description = "d" -> "a ipv4"
              ~ network     = "x.x.x.x/32" -> "x.x.x.x/23"
            }
          ~ ip_filter_object {
              ~ description = "c" -> "a ipv6"
              ~ network     = "x.x.x.x/16" -> "x:x:x::/48"
            }
          ~ ip_filter_object {
              ~ description = "a ipv6" -> "c"
              ~ network     = "x:x:x::/48" -> "x.x.x.x/16"
            }
          ~ ip_filter_object {
              ~ description = "a ipv4" -> "d"
              ~ network     = "x.x.x.x/23" -> "x.x.x.x/32"
            }
```

And then after a while, again without touching the terraform files:
```
          ~ ip_filter_object {
              ~ description = "d" -> "a ipv4"
              ~ network     = "x.x.x.x/32" -> "x.x.x.x/23"
            }
          ~ ip_filter_object {
              ~ description = "c" -> "a ipv6"
              ~ network     = "x.x.x.x/16" -> "x:x:x::/48"
            }
          ~ ip_filter_object {
              ~ description = "a ipv6" -> "c"
              ~ network     = "x:x:x::/48" -> "x.x.x.x/16"
            }
          ~ ip_filter_object {
              ~ description = "a ipv4" -> "d"
              ~ network     = "x.x.x.x/23" -> "x.x.x.x/32"
            }
```

## What did you expect to happen?

ip_filter_objects to stay in the same order when nothing changes

## What else do we need to know?

Version 4.14.0

### PR DESCRIPTION
**Pull request by byashimov**
## About this change—what it does

Set type is the only way to fix `ip_filter_object` field.

- renames `hasValue` to `isNull` in `converters.go`, as it missleading
- uses `TypeSet` for `ip_filter_object`

Resolves #1638, #1107.

### CODE REVIEW

**Comment by byashimov** [comment]
tests
- https://github.com/aiven/terraform-provider-aiven/actions/runs/8266608999
- https://github.com/aiven/terraform-provider-aiven/actions/runs/8265421038

File: internal/sdkprovider/userconfig/diff/diff.go [hunk 1]
```diff
@@ -64,3 +73,25 @@ func suppressIPFilterSet(k, old, new string, d *schema.ResourceData) bool {
...
+
+	// user_config field
+	path = path[1:]
+
+	// Drills down the field
+	for _, v := range path {
+		if v == "0" {
+			value = value.AsValueSlice()[0]
+		}
+		if value.Type().HasAttribute(v) {
```
Comment (byashimov) [inline internal/sdkprovider/userconfig/diff/diff.go:90]: this must be early exit, going to fix now

File: internal/sdkprovider/userconfig/converters/converters.go [hunk 2]
```diff
@@ -23,7 +23,10 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
-const userConfigSuffix = "_user_config"
+const (
+	userConfigSuffix   = "_user_config"
+	AllowIPFilterPurge = "AIVEN_ALLOW_IP_FILTER_PURGE"
```
Comment (Serpentiel) [inline internal/sdkprovider/userconfig/converters/converters.go:28]: Why do we make it an exported const? I don't know if it's a good idea if we only use it in the test, maybe the test should also define the const? WDYT?

**Review by Serpentiel — COMMENTED** [review]
lgtm, but I have a question

File: internal/sdkprovider/userconfig/converters/converters.go (hunk 2 above)
Comment (byashimov) [reply internal/sdkprovider/userconfig/converters/converters.go:28]: `AllowIPFilterPurge` is used to format the error message too.

Comment (Serpentiel) [reply internal/sdkprovider/userconfig/converters/converters.go:28]: Yes, I can see, but I mean why is it an exported const? We can make it a non-exported const instead, and introduce a separate one in the test, WDYT?

Comment (byashimov) [reply internal/sdkprovider/userconfig/converters/converters.go:28]: The idea of constants is to be the source of truth.

**Comment by byashimov** [comment]
@Serpentiel I have restarted the job, and that's reseted the approve. Could you please approve it again?

**Review by Serpentiel — APPROVED** [review]

**Pull request merged**